api/support/verify_state.py

```
@@ -0,0 +1,265 @@
+        return []
+
+    block_hex = chain_state["old_block"]
+    addr_hex = PROBE_ADDRESSES[chain]
+    addr_bytes = bytes.fromhex(addr_hex.removeprefix("0x"))
+    ts_ns = time.time_ns()
+
+    providers = all_providers_for(chain)
+    chainstack_url = chainstack_endpoint_for(chain)
+
+    if not providers or not chainstack_url:
+        logging.warning(f"verify_state: missing endpoints for {chain}")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)